File: app/src/main/java/io/intelehealth/client/offline_login/NetworkConnection.java
package io.intelehealth.client.offline_login;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

/**
 * Created by Dexter Barretto on 5/24/17.
 * Github : @dbarretto
 */

public class NetworkConnection {

    public static boolean isOnline(Context context) {
        ConnectivityManager cm =
                (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo netInfo = cm.getActiveNetworkInfo();
        return netInfo != null && netInfo.isConnected();
    }

}

File: features/ondemand/abdm/src/main/java/org/intelehealth/abdm/util/NetworkConnection.kt
package org.intelehealth.abdm.util

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkCapabilities

/**
 * Lightweight connectivity check used to guard the ABHA flow entry (legacy parity).
 * Self-contained so the module carries no dependency on the host's network utilities.
 */
object NetworkConnection {

    fun isOnline(context: Context): Boolean {
        val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as? ConnectivityManager
            ?: return false
        val network = cm.activeNetwork ?: return false
        val capabilities = cm.getNetworkCapabilities(network) ?: return false
        return capabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) ||
            capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)
    }
}
